test/penalizer/Penalizer.test.ts

```diff
@@ -0,0 +1,366 @@
+        to: recipient.address,
+        relayData: '0xdeadbeef05',
+        enableQos: true
+      })
+      const sig = relayHelper.getRelayRequestSignature(rr, sender)
+
+      const receipt = relayHelper.createReceipt({ relayRequest: rr, signature: sig })
+
+      await expectRevert(
+        penalizer.claim(receipt),
+        'worker signature mismatch'
+      )
+    })
+
+    it('due to forged signature', async function () {
+      const rr = await relayHelper.createRelayRequest({
+        from: sender.address,
+        to: recipient.address,
+        relayData: '0xdeadbeef06',
+        enableQos: true
+      })
+      const sig = relayHelper.getRelayRequestSignature(rr, sender)
+
+      const receipt = relayHelper.createReceipt({ relayRequest: rr, signature: sig })
+      receipt.workerSignature = web3.utils.randomHex(130)
+
+      await expectRevert(
+        penalizer.claim(receipt),
+        'worker signature mismatch'
+      )
+    })
+
+    it('due to worker address mismatch', async function () {
+      const rr = await relayHelper.createRelayRequest({
+        from: sender.address,
+        to: recipient.address,
+        relayData: '0xdeadbeef07',
+        enableQos: true
+      })
+      const sig = relayHelper.getRelayRequestSignature(rr, sender)
+
+      const receipt = relayHelper.createReceipt({ relayRequest: rr, signature: sig })
+      receipt.commitment.relayWorker = otherAccount
+      await relayHelper.signReceipt(receipt)
+
+      await expectRevert(
+        penalizer.claim(receipt),
+        'worker address does not match'
+      )
+    })
+
+    it('due to relay hub mismatch', async function () {
+      const rr = await relayHelper.createRelayRequest({
+        from: sender.address,
+        to: recipient.address,
+        relayData: '0xdeadbeef07',
+        enableQos: true
+      })
+      const sig = relayHelper.getRelayRequestSignature(rr, sender)
+
+      const receipt = relayHelper.createReceipt({ relayRequest: rr, signature: sig })
+      receipt.commitment.relayHubAddress = otherAccount
+      await relayHelper.signReceipt(receipt)
+
+      await expectRevert(
+        penalizer.claim(receipt),
+        'relay hub does not match'
+      )
+    })
+
+    it('due to claim not being made by commitment receiver', async function () {
+      const rr = await relayHelper.createRelayRequest({
+        from: sender.address,
+        to: recipient.address,
+        relayData: '0xdeadbeef08',
+        enableQos: true
+      })
+      const sig = relayHelper.getRelayRequestSignature(rr, sender)
+
+      const receipt = relayHelper.createReceipt({ relayRequest: rr, signature: sig })
+      await relayHelper.signReceipt(receipt)
+
+      await expectRevert(
+        penalizer.claim(receipt, { from: otherAccount }),
+        'receiver must claim commitment'
+      )
+    })
+  })
+
+  describe('should receive claims', function () {
+    before(async function () {
+      // sender will need to be funded to complete claim calls
```

Comment:
we could in theory _also_ relay `claim` calls, but i think this is just too convoluted for these tests.

i suppose we could include this in a later iteration